src/Tabs/Tab.tsx

```diff
@@ -4,18 +4,23 @@ import { StyledTab, StyledTabWrapper } from './Tab.styles';
 
 export interface TabProps {
   tabId: string;
-  beforeOnChange?: () => boolean;
+  beforeOnChange?: (handleClick: () => void) => boolean;
 }
 
 export const Tab: React.FC<TabProps> = ({ tabId, beforeOnChange, children, ...rest }) => {
   const { setActiveTab, activeTab } = useTabsContext();
-  const handleClick = (): void => {
+  const clickCallback = (): void => {
     if (activeTab === tabId) return;
-
-    const shouldChange = beforeOnChange?.();
-    if (beforeOnChange && !shouldChange && shouldChange !== undefined) return;
     setActiveTab(tabId);
   };
+
+  const handleClick = (): void => {
+    if (beforeOnChange) {
+      beforeOnChange(clickCallback);
+    } else {
+      clickCallback();
+    }
```

Comment:
Not a big deal, but if there is only one thing in both if and else, should it be a one-liner? `beforeOnChange ? beforeOnChange(clickCallback) : clickCallback();`